packages/dev/core/test/unit/Math/babylon.math.vector.test.ts

```diff
@@ -1,61 +1,1240 @@
+        beforeEach(() => {
+            vector1 = new Vector2(1, 2);
+            vector2 = new Vector2(3, 4);
+        });
+
+        describe("add", () => {
+            it("should add two vectors and return a new vector", () => {
+                const result = vector1.add(vector2);
+                expect(result.x).toBe(4);
+                expect(result.y).toBe(6);
+            });
+        });
+
+        describe("addInPlace", () => {
+            it("should add the given vector to the current vector", () => {
+                vector1.addInPlace(vector2);
+                expect(vector1.x).toBe(4);
+                expect(vector1.y).toBe(6);
+            });
+        });
+
+        describe("addToRef", () => {
+            it("should add the given vector to the current vector and store the result in the given output vector", () => {
+                const result = new Vector2();
+                vector1.addToRef(vector2, result);
+                expect(result.x).toBe(4);
+                expect(result.y).toBe(6);
+            });
+        });
+        describe("addVector3", () => {
+            it("should add a Vector3 to the current vector", () => {
+                const vector3 = new Vector3(5, 6, 7);
+                const result = vector1.addVector3(vector3);
+                expect(result.x).toBe(6);
+                expect(result.y).toBe(8);
+            });
+        });
+
+        describe("asArray", () => {
+            it("should return an array representation of the vector", () => {
+                const result = vector1.asArray();
+                expect(result.length).toBe(2);
+                expect(result[0]).toBe(1);
+                expect(result[1]).toBe(2);
+            });
+        });
+
+        describe("clone", () => {
+            it("should create a new vector with the same values", () => {
+                const cloned = vector1.clone();
+                expect(cloned).not.toBe(vector1);
+                expect(cloned.x).toBe(vector1.x);
+                expect(cloned.y).toBe(vector1.y);
+            });
+        });
+
+        describe("copyFrom", () => {
+            it("should copy the values from another vector", () => {
+                vector1.copyFrom(vector2);
+                expect(vector1.x).toBe(vector2.x);
+                expect(vector1.y).toBe(vector2.y);
+            });
+        });
+
+        describe("copyFromFloats", () => {
+            it("should set the vector from individual x and y values", () => {
+                vector1.copyFromFloats(5, 6);
+                expect(vector1.x).toBe(5);
+                expect(vector1.y).toBe(6);
+            });
+        });
+
+        describe("divide", () => {
+            it("should divide the vector by another vector", () => {
+                const result = vector1.divide(vector2);
+                expect(result.x).toBeCloseTo(1 / 3);
+                expect(result.y).toBeCloseTo(2 / 4);
+            });

```

Comment:
Same, checking if vector1 and vector2 are not changed would be nice

Reply:
Added in [75a2f6c](https://github.com/BabylonJS/Babylon.js/pull/13973/commits/75a2f6c7681087089282cc29d62f97520f590ea8)